prime-router/src/main/kotlin/azure/ApiKeysFunctions.kt

```diff
@@ -15,8 +15,57 @@ import gov.cdc.prime.router.tokens.AuthenticatedClaims
 import gov.cdc.prime.router.tokens.JwkSet
 import gov.cdc.prime.router.tokens.Scope
 import gov.cdc.prime.router.tokens.authenticationFailure
+import io.swagger.v3.oas.annotations.OpenAPIDefinition
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.Parameter
+import io.swagger.v3.oas.annotations.enums.SecuritySchemeType
+import io.swagger.v3.oas.annotations.info.Contact
+import io.swagger.v3.oas.annotations.info.Info
+import io.swagger.v3.oas.annotations.media.Content
+import io.swagger.v3.oas.annotations.media.Schema
+import io.swagger.v3.oas.annotations.responses.ApiResponse
+import io.swagger.v3.oas.annotations.security.OAuthFlow
+import io.swagger.v3.oas.annotations.security.OAuthFlows
+import io.swagger.v3.oas.annotations.security.OAuthScope
+import io.swagger.v3.oas.annotations.security.SecurityRequirement
+import io.swagger.v3.oas.annotations.security.SecurityScheme
 import org.apache.logging.log4j.kotlin.Logging
+import javax.ws.rs.DELETE
+import javax.ws.rs.GET
+import javax.ws.rs.POST
+import javax.ws.rs.Path
+import javax.ws.rs.PathParam
 
+@OpenAPIDefinition(
```

Comment:
I'm a bit confused. Will each azure function/set of endpoints have its own .yml file? We should figure out how to get them all in one OR figure out how to factor this general APIDefinition out, because it is not specific to these endpoints?

If each azure function will have its own file, we should also not name the output file api.yml

Reply:
there should be ways to consolidate yamls into one page and divided into sections etc.

Will look into it

I don't do this docs thing often, so need to look around

Reply:
this ticket is a pilot on ApiKeysFunctions.kt, and we could layout the schemes for a later task that generate all API endpoints:

the yaml file name can be any name we want (they won't show on final swagger UI served), e.g. lower case the kotlin file name that contains the end points, or whatever the naming scheme.

I think the generator can be configured to group the end points across the source files (*.kt) and generate a set of specs - yaml files, and the multiple yaml files can be configured into Swagger UI runtime, by e.g. see the js code below:

window.onload = function() {
  // Build a system
  const ui = SwaggerUIBundle({
    urls: [
      {url: "https://path/to/rs_api_grp1.yaml", name: "API for Key Managements"},
      {url: "https://path/to/rs_api_grp2.yaml", name: "API Report Submission History"},
    ],
    "urls.primaryName": "API Two"  // default document (if other than the first)
    ...
  })

Reply:
Not in AC of this ticket

Can be done in the ticket for generate specs for all APIs endpoints and deployment (serving) task.

Reply:
the visual looks like, endpoints in different yaml files are considered different API (or sub APIs), and can be selected from swagger UI top bar pull down list:

![image](https://github.com/CDCgov/prime-reportstream/assets/135686833/9bdf682f-b273-4081-a546-15b599b8b707)